### DISCUSSION TITLE
Cannot get Azure DevOps Pull Requests to work

### DISCUSSION THREAD

**Discussion by Mathyn** [WeblateOrg/weblate#12002] [Q&A]
Hi,

I am working on setting up Weblate to use in combination with Azure DevOps. I've set things up so Weblate pulls changes from the `develop` branch and will also push changes back to the `develop` branch.

Since I use pull requests to merge work in `develop` I've locked down the `develop` branch to only allow merging from a pull request. This means Weblate must also create pull requests when it wants to push changes.

I've found [Weblate documentation](https://docs.weblate.org/en/latest/vcs.html#azure-devops-pull-requests) on how to set this up and managed to configure things so the 'Azure DevOps pull request' option shows up when configuring the Version Control System.

In short to set things up I've configured the following environment variables (real values replaced with dummy values):

```
WEBLATE_AZURE_DEVOPS_USERNAME=MyProjectName
WEBLATE_AZURE_DEVOPS_ORGANIZATION=MyOrganizationName
WEBLATE_AZURE_DEVOPS_TOKEN=MyToken
WEBLATE_AZURE_DEVOPS_HOST=dev.azure.com
```

However as soon as I click the 'Push' button in 'Repository Maintenance' I get the following error:

```
Could not fork repository at dev.azure.com: Repository not found. Check whether exists and MyProjectName has access to it.
```

This seems to suggest two things:
- Between 'whether' and 'exists' it seems like the project name should be set, however I cannot see any way to configure this.
- `WEBLATE_AZURE_DEVOPS_USERNAME` according to the [documentation](https://docs.weblate.org/en/latest/admin/config.html#std-setting-AZURE_DEVOPS_CREDENTIALS) should be the project name, but this error message suggests it should be a user name?

Anyone have any success using the pull request feature from Azure DevOps?

I am using a self hosted version of Weblate 5.6.2

**Comment by nijel** [comment]
I have no experience with Azure DevOps, but there was some discussion on this at https://github.com/WeblateOrg/weblate/pull/10227

**Reply by Mathyn** [reply]
Thanks @nijel I'll take a look, some comments certainly seem to apply to what I am seeing. I'll leave this discussion open for now, but will update as soon as I get (or find) an answer.
